code/game/objects/items/weapons/weaponry.dm

```diff
@@ -5,12 +5,16 @@
 /obj/item/weapon/examine(mob/user)
 	..()
 	if(unique_rename)
-		user << "<span class='notice'>Use a pen on it to rename it.</span>"
+		user << "<span class='notice'>Use a pen on it to rename it or change its description.</span>"
 
 /obj/item/weapon/attackby(obj/item/I, mob/user, params)
 	if(unique_rename)
 		if(istype(I, /obj/item/weapon/pen))
-			rename_weapon(user)
+			var/penchoice = alert("What would you like to edit?", "Rename or change description?", "Rename", "Change description", "Cancel")

```

Comment:
This is a sleep so you need

    if(!qdeleted(src) && canUseTopic(src, be_close = TRUE))